packages/@aws-cdk/aws-kinesisfirehose-alpha/lib/delivery-stream.ts

```diff
@@ -328,17 +331,20 @@ export class DeliveryStream extends DeliveryStreamBase {
       });
     }
 
+    const encryption = props.encryption ?? StreamEncryption.awsOwnedKey();
+    const encryptionKey = encryption.encryptionKey ?? (encryption.type === StreamEncryptionType.CUSTOMER_MANAGED ? new kms.Key(this, 'Key') : undefined);
```

Comment:
This `encryptionKey` feels a bit off to me. Generally if it's a customer managed key, I believe we should allow users to be able to specify a key themselve instead of always creating a key for them.

Reply:
For customer managed key, passing in their own key is optional so this only creates a key when customer managed key is set if they didn't pass in their own, otherwise `const encryptionKey = encryption.encryptionKey` would use the key provided.